distribution/push.go

```diff
@@ -126,3 +153,32 @@ func compress(in io.Reader) (io.ReadCloser, chan struct{}) {
 
 	return pipeReader, compressionDone
 }
+
+func pigzCompress(ctx context.Context, in io.Reader, pigzPath string) (io.ReadCloser, chan struct{}) {
+	cmd := exec.CommandContext(ctx, pigzPath, "-c")
+
+	cmd.Stdin = in
+	pipeR, pipeW := io.Pipe()
+	cmd.Stdout = pipeW
+	var errBuf bytes.Buffer
+	cmd.Stderr = &errBuf
+
+	done := make(chan struct{})
+
+	if err := cmd.Start(); err != nil {
+		fmt.Errorf("could not get compression stream: %v", err)
+		close(done)
+		return nil, done
+	}
+
+	go func() {
+		if err := cmd.Wait(); err != nil {
+			pipeW.CloseWithError(fmt.Errorf("%s: %s", err, errBuf.String()))
+		} else {
+			pipeW.Close()
+		}
+		close(done)
+	}()
+
+	return pipeR, done
+}
```

Comment:
Oh, while you're updating; could you also fix the missing newline here? 😅

Reply:
I'm very sorry. I will fix it in the next commit. (By the way, if I commit now, will the build restart?)

Reply:
No worries! Yes, amending a commit, and force-pushing should restart the build. As you're a first-time contributor, I still need to manually approve GitHub Actions (but just did, so it should be running now)

Reply:
Thanks for the guidance, it looks like the checks have passed.